easybuild/tools/filetools.py

```diff
@@ -1787,9 +1787,21 @@ def get_source_tarball_from_git(filename, targetdir, git_config):
 
         run.run_cmd(' '.join(checkout_cmd), log_all=True, log_ok=False, simple=False, regexp=False, path=repo_name)
 
-    # create an archive and delete the git repo directory
-    tar_cmd = ['tar', 'cfvz', targetpath, '--exclude', '.git', repo_name]
-    run.run_cmd(' '.join(tar_cmd), log_all=True, log_ok=False, simple=False, regexp=False)
+    # create a git archive (in tar format, gzip later)
+    git_cmd = ['git', 'archive', '-o', targetpath[:-3], '--prefix=%s/' % (repo_name), 'HEAD']
+    run.run_cmd(' '.join(git_cmd), log_all=True, log_ok=False, simple=False, regexp=False, path=repo_name)
+
+    # git archive does not automatically recurse into submodules
+    if recursive:
+        # create a git archive for each submdule, and append it to the target
+        git_cmd = ['git', 'submodule', 'foreach', "'git archive -o $name.tar --prefix=%s/$path/ HEAD'" % (repo_name)]
+        tar_cmd = ['git', 'submodule', 'foreach', "'tar -f %s --concatenate $name.tar'" % (targetpath[:-3])]
```

Comment:
There is a potential problem here, if the submodule info is changed so that the order of the submodules changes without changing anything else, the resulting concatenated tar file will be different.
It would be better to pick the submodules list out, sort it and then do archive/tar

Reply:
I wasn't sure how git submodule foreach worked. But it is going in LANG=C sort order regardless of .gitmodule. So this is not a propblem.